README.md

```diff
@@ -1,2 +1,79 @@
-# devfile-sample-java-springboot-basic
-A basic sample application using Java Spring Boot with devfile
+# sample-devfile-with-basic-java-spring-boot
+
+By default, the Spring Boot application uses the **HTTP 8080** port. You can use a devfile to define how Spring Boot interacts with your application.
+
+## Running Spring Boot from the command line
+
+You can run Spring Boot with the `java -jar` command, like this:
```

Comment:
These files are in this repository https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/devfile.yaml and https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/docker/Dockerfile

Reply:
Re "Checkout the file docker/Dockerfile on how to build this application and containerize it":

https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/docker/Dockerfile contains the logic to build this repo springboot project. So your CLI commands that you would run on your laptop environment for instance, are essentially captured in the Dockerfile. 

So for instance, you raised the topic regarding if you would need the pom.xml file and executing the CLI java -jar.... They would be captured in this Dockerfile. 

Since RHTAP and OpenShift Console(ODC) are the prime consumers of this sample, they would not execute CLI outside the Dockerfile. They would execute the Dockerfile instructions. So I dont think it is necessary to make a note of it separately. I guess we are covering this topic anyways with my point "Checkout the file docker/Dockerfile on how to build this application and containerize it"

Does that sound good to you?

Reply:
it does. thanks!

Reply:
Sorry, ".. build this _sample/project_ as an image"

It is the same image. To clarify from the beginning:

1. [docker/Dockerfile](https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/docker/Dockerfile) has instructions on how to build this sample as a container image instead of CLI
2. [devfile.yaml image component](https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/devfile.yaml#L21-L27) has a reference to the Dockerfile from point 1 above 
3. Once the image has been built using the Dockerfile, then to deploy the built image for this sample/project; we use a [devfile.yaml kubernetes component](https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/devfile.yaml#L28-L41) that tells us how to deploy
4. Now to chain the points 2 and 3 together, we use [devfile.yaml apply and deploy commands](https://github.com/devfile-samples/devfile-sample-java-springboot-basic/blob/main/devfile.yaml#L43-L56), these commands reference the devfile.yaml components mentioned in point 2 and 3 above.

The outerloop phase is basically these points